src/Adapter/Product/CommandHandler/DeleteProductSupplierHandler.php

```diff
@@ -29,68 +29,69 @@
 use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
-use PrestaShopException;
-use ProductSupplier;
 
 /**
  * Handles @var DeleteProductSupplierCommand using legacy object model
  */
 final class DeleteProductSupplierHandler extends AbstractProductSupplierHandler implements DeleteProductSupplierHandlerInterface
 {
     /**
-     * {@inheritdoc}
+     * @var ProductSupplierDeleterInterface
      */
-    public function handle(DeleteProductSupplierCommand $command): void
-    {
-        $productSupplier = $this->getProductSupplier($command->getProductSupplierId());
+    private $productSupplierDeleter;
 
-        try {
-            if (!$productSupplier->delete()) {
-                throw new CannotDeleteProductSupplierException(sprintf(
-                    'Failed deleting product supplier #%d',
-                    $productSupplier->id
-                ));
-            }
-            $this->refreshProductDefaultSupplier($productSupplier);
-        } catch (PrestaShopException $e) {
-            throw new ProductSupplierException(
-                sprintf('Error occurred when deleting product supplier #%d', $productSupplier->id),
-                0,
-                $e
-            );
-        }
+    /**
+     * @var ProductProvider
+     */
+    private $productProvider;
+
+    /**
+     * @var ProductSupplierProvider
+     */
+    private $productSupplierProvider;
+
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @param ProductSupplierDeleterInterface $productSupplierDeleter
+     * @param ProductProvider $productProvider
+     * @param ProductSupplierProvider $productSupplierProvider
+     * @param ProductUpdater $productUpdater
+     */
+    public function __construct(
+        ProductSupplierDeleterInterface $productSupplierDeleter,
+        ProductProvider $productProvider,
+        ProductSupplierProvider $productSupplierProvider,
+        ProductUpdater $productUpdater
+    ) {
+        $this->productSupplierDeleter = $productSupplierDeleter;
+        $this->productProvider = $productProvider;
+        $this->productSupplierProvider = $productSupplierProvider;
+        $this->productUpdater = $productUpdater;
     }
 
     /**
-     * @param ProductSupplier $productSupplier
-     *
-     * @throws CannotUpdateProductException
-     * @throws ProductException
-     * @throws ProductNotFoundException
+     * {@inheritdoc}
      */
-    private function refreshProductDefaultSupplier(ProductSupplier $productSupplier): void
+    public function handle(DeleteProductSupplierCommand $command): void
     {
-        $productId = (int) $productSupplier->id_product;
-        $product = $this->getProduct(new ProductId($productId));
+        $this->productSupplierDeleter->delete($command->getProductSupplierId());
+        $productSupplier = $this->productSupplierProvider->get($command->getProductSupplierId());
```

Comment:
I don't understand you get the ProductSupplier after it's been removed, shouldn't it be the opposite Fetch before it's been removed or else you can't fetch it any more logically

Reply:
This made me realize that this command is not behat tested by the way

Reply:
indeed this command is full of crap :sweat:

Reply:
The crap of a developer is the treasure of another (they say)